pkg/ingester/client/compat.go

```diff
@@ -229,17 +229,21 @@ func FromLabelAdaptersToLabels(ls []LabelAdapter) labels.Labels {
 // get in input labels whose data structure is reused.
 func FromLabelAdaptersToLabelsWithCopy(input []LabelAdapter) labels.Labels {
 	result := make(labels.Labels, len(input))
-
 	for i, l := range input {
 		result[i] = labels.Label{
-			Name:  l.Name,
-			Value: l.Value,
+			Name:  copyString(l.Name),
+			Value: copyString(l.Value),
 		}
 	}
-
 	return result
 }
 
+func copyString(in string) string {
```

Comment:
In pkg/util/extract/extract.go we do ` string([]byte(label.Value))`

Reply:
I know, I was just afraid that Go compiler can easily optimize that away as no-op.

Reply:
Also in `pkg/ingester/index/index.go` yay DRY.